src/simsopt/field/coil.py

```diff
@@ -118,6 +122,45 @@ def vjp(self, v_current):
     def get_value(self):
         return self.scale * self.current_to_scale.get_value()
 
+    def set_dofs(self, dofs):
+        self.current_to_scale.set_dofs(dofs / self.scale)
+
+
+def current_pure(dofs):
+    return dofs
+
+
+class JaxCurrent(sopp.Current, CurrentBase):
+    def __init__(self, current, dofs=None, **kwargs):
+        sopp.Current.__init__(self, current)
+        if dofs is None:
+            CurrentBase.__init__(self, external_dof_setter=sopp.Current.set_dofs,
+                                 x0=self.get_dofs(), **kwargs)
+        else:
+            CurrentBase.__init__(self, external_dof_setter=sopp.Current.set_dofs,
+                                 dofs=dofs, **kwargs)
+
+        self.current_pure = current_pure
```

Comment:
Seems unnecessary to assign current_pure to self. Can just use current_pure right?